core/src/main/java/org/elasticsearch/index/mapper/UidFieldMapper.java

```diff
@@ -110,19 +110,15 @@ public String typeName() {
         }
 
         @Override
-        public IndexFieldData.Builder fielddataBuilder() {
+        public IndexFieldData.Builder fielddataBuilder(String fullyQualifiedIndexName) {
             if (indexOptions() == IndexOptions.NONE) {
                 DEPRECATION_LOGGER.deprecated("Fielddata access on the _uid field is deprecated, use _id instead");
-                return new IndexFieldData.Builder() {
-                    @Override
-                    public IndexFieldData<?> build(IndexSettings indexSettings, MappedFieldType fieldType, IndexFieldDataCache cache,
-                            CircuitBreakerService breakerService, MapperService mapperService) {
-                        MappedFieldType idFieldType = mapperService.fullName(IdFieldMapper.NAME);
-                        IndexFieldData<?> idFieldData = idFieldType.fielddataBuilder()
-                                .build(indexSettings, idFieldType, cache, breakerService, mapperService);
-                        final String type = mapperService.types().iterator().next();
-                        return new UidIndexFieldData(indexSettings.getIndex(), type, idFieldData);
-                    }
+                return (indexSettings, fieldType, cache, breakerService, mapperService) -> {
```

Comment:
nit: when the method is complex (there are 5 different arguments here), I find that explicitly implementing the interface is easier to read than lambdas